app/app_test.go

```diff
@@ -45,39 +53,115 @@ func TestEvmosExport(t *testing.T) {
 	require.NoError(t, err)
 
 	// Initialize the chain
-	app.InitChain(
+	evmosApp.InitChain(
 		abci.RequestInitChain{
 			ChainId:       chainID,
 			Validators:    []abci.ValidatorUpdate{},
 			AppStateBytes: stateBytes,
 		},
 	)
-	app.Commit()
+	evmosApp.Commit()
 
 	// Making a new app object with the db, so that initchain hasn't been called
-	app2 := NewEvmos(
+	app2 := app.NewEvmos(
 		log.NewTMLogger(log.NewSyncWriter(os.Stdout)),
 		db, nil, true, map[int64]bool{},
-		DefaultNodeHome, 0,
-		encoding.MakeConfig(ModuleBasics),
-		simtestutil.NewAppOptionsWithFlagHome(DefaultNodeHome),
+		app.DefaultNodeHome, 0,
+		encoding.MakeConfig(app.ModuleBasics),
+		simtestutil.NewAppOptionsWithFlagHome(app.DefaultNodeHome),
 		baseapp.SetChainID(chainID),
 	)
 	_, err = app2.ExportAppStateAndValidators(false, []string{}, []string{})
 	require.NoError(t, err, "ExportAppStateAndValidators should not have an error")
 }
+
+// This test checks is a safeguard to avoid missing precompiles that should be blocked addresses.
+//
+// It does so, by initially expecting all precompiles available in the EVM to be blocked, and
+// require developers to specify exactly which should be an exception to this rule.
+func TestPrecompilesAreBlockedAddrs(t *testing.T) {
+	keyring := testkeyring.New(1)
+	signer := keyring.GetKey(0)
+	network := testnetwork.NewUnitTestNetwork(
+		testnetwork.WithPreFundedAccounts(signer.AccAddr),
+	)
+	handler := grpc.NewIntegrationHandler(network)
+	factory := testfactory.New(network, handler)
+
+	// NOTE: all precompiles that should NOT be blocked addresses need to go in here
+	precompilesAbleToReceiveFunds := []ethcommon.Address{
+		ethcommon.HexToAddress("0x0000000000000000000000000000000000000001"),
+		ethcommon.HexToAddress("0x0000000000000000000000000000000000000002"),
+		ethcommon.HexToAddress("0x0000000000000000000000000000000000000003"),
+		ethcommon.HexToAddress("0x0000000000000000000000000000000000000004"),
+		ethcommon.HexToAddress("0x0000000000000000000000000000000000000005"),
+		ethcommon.HexToAddress("0x0000000000000000000000000000000000000006"),
+		ethcommon.HexToAddress("0x0000000000000000000000000000000000000007"),
+		ethcommon.HexToAddress("0x0000000000000000000000000000000000000008"),
+		ethcommon.HexToAddress("0x0000000000000000000000000000000000000009"),
+	}
+
+	precompilesAbleToReceiveFundsViaEth := []ethcommon.Address{
+		ethcommon.HexToAddress("0x0000000000000000000000000000000000000001"),
+		ethcommon.HexToAddress("0x0000000000000000000000000000000000000002"),
+		ethcommon.HexToAddress("0x0000000000000000000000000000000000000003"),
+		ethcommon.HexToAddress("0x0000000000000000000000000000000000000004"),
+		ethcommon.HexToAddress("0x0000000000000000000000000000000000000005"),
+		ethcommon.HexToAddress("0x0000000000000000000000000000000000000006"),
+		ethcommon.HexToAddress("0x0000000000000000000000000000000000000007"),
+		ethcommon.HexToAddress("0x0000000000000000000000000000000000000008"),
```

Comment:
Interestingly the `0x09` precompile also doesn't work for the simple ETH transfer via an EVM transaction - other than sending via Cosmos transaction to this address.